ospackage/recipes/steps.go

```diff
@@ -84,39 +64,51 @@ func StepFileExec(step *osconfigpb.SoftwareRecipe_Step_FileExec, artifacts map[s
 		var ok bool
 		path, ok = artifacts[v.ArtifactId]
 		if !ok {
-			return nil, fmt.Errorf("%q not found in artifact map", v.ArtifactId)
+			return fmt.Errorf("%q not found in artifact map", v.ArtifactId)
 		}
 	default:
-		return nil, fmt.Errorf("can't determine location type")
+		return fmt.Errorf("can't determine location type")
 	}
 
-	res := []string{path}
-	res = append(res, step.FileExec.Args...)
-	return res, nil
+	return executeCommand(path, stepDir, runEnvs, step.FileExec.Args...)
 }
 
 // StepScriptRun builds the command for a ScriptRun step
-func StepScriptRun(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, artifacts map[string]string) ([]string, error) {
-	// TODO: should be putting this in stepN_type/ dir, but that needs me to know the dir way in advance..
-	// actually this is an artifact. we should have made them referenced as artifacts.. we'll have to repeat file creation logic here.
-	f, err := os.Create("/tmp/scriptrun")
+func StepScriptRun(step *osconfigpb.SoftwareRecipe_Step_ScriptRun, artifacts map[string]string, runEnvs []string, stepDir string) error {
+	scriptPath := filepath.Join(stepDir, "script")
+
+	f, err := os.Create(scriptPath)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	defer f.Close()
 
 	f.WriteString(step.ScriptRun.Script)
-	f.Sync()
+	f.Close()
 	if err := os.Chmod("/tmp/scriptrun", 0755); err != nil {
-		return nil, err
+		return err
 	}
 
-	res := []string{"/bin/sh", "-c"}
-	//if step.ScriptRun.Interpreter == osconfigpb.SoftwareRecipe_Step_RunScript_POWERSHELL {
 	var qargs []string
+	//if step.ScriptRun.Interpreter == osconfigpb.SoftwareRecipe_Step_RunScript_POWERSHELL {
 	for _, arg := range step.ScriptRun.Args {
 		qargs = append(qargs, fmt.Sprintf("%q", arg))
 	}
-	res = append(res, "/tmp/scriptrun"+" "+strings.Join(qargs, " "))
-	return res, nil
+	command := scriptPath + " " + strings.Join(qargs, " ")
+	args := []string{"-c", command}
+	return executeCommand("/bin/sh", stepDir, runEnvs, args...)
+}
+
+func executeCommand(cmd string, workDir string, envs []string, args ...string) error {
+	cmdObj := exec.Command(cmd, args...)
+
+	cmdObj.Dir = workDir
+	cmdObj.Env = append(cmdObj.Env, envs...)
+	cmdObj.Env = append(cmdObj.Env, fmt.Sprintf("PWD=%s", cmdObj.Dir))
```

Comment:
PWD is set by cmdObj.Dir and shouldn't be needed here

Reply:
Done.